### DISCUSSION TITLE
Custom Icons / Buttons on Smoothstep / Step Edge Segment

### DISCUSSION THREAD

**Discussion by mahesh1402** [xyflow/xyflow#1727] [General]
I am working on a project where I want to Populate Yes / No Buttons on smooth step / step edge's start segment . Something like :

![EdgesYN](https://user-images.githubusercontent.com/62994341/143811427-7e1678fa-cb23-490f-be2f-457c7ff6f763.jpg)

How can I obtain Edge Direction and segment position to display icons / buttons on start of the edge that is outgoing from node ?
Also want to toggle Yes/No buttons by clicking on it. Icon / Button need to be populated immediately after few pixel of starting of edge before it turns to any direction. Would like to get if Edge Going Up/Down in Y Direction or Left / Right in X Position. May be is there any better way to do this ?

**Comment by datoslabs** [comment]
Yes, you can implement all the logic in a custom edge.

**Reply by samuel-akinboro** [reply]
how
